src/utils/viewer/viewer.cpp

```diff
@@ -388,12 +394,15 @@ Viewer::Viewer(const Arguments& arguments)
     debugBullet_ = true;
   }
 
+  gfxReplayRecordFilepath_ = args.value("gfx-replay-record-filepath");
+
   // configure and intialize Simulator
   auto simConfig = esp::sim::SimulatorConfiguration();
   simConfig.activeSceneID = sceneFileName;
   simConfig.enablePhysics = useBullet;
   simConfig.frustumCulling = true;
   simConfig.requiresTextures = true;
+  simConfig.enableGfxReplaySave = !gfxReplayRecordFilepath_.empty();
```

Comment:
If this is false, `simulator_->getGfxReplayManager()->getRecorder()` returns nullptr?

Reply:
yes (er, null shared_ptr)